### DISCUSSION TITLE
Python Inference

### DISCUSSION THREAD

**Discussion by npiroozan** [deepmodeling/deepmd-kit#1841] [Q&A]
Hello,

I hope you are doing well!

Forgive me if this is a bit of a foolish question on my part but I'm getting an error using Inference with Python API that I'm not understanding well.

The code I'm using is the following:

from deepmd.infer import DeepPot
import numpy as np
dp = DeepPot('graph.pb')
coord = np.array([[1,0,0], [0,0,1.5], [1,0,3]]).reshape([1, -1])
cell = np.diag(10 * np.ones(3)).reshape([1, -1])
atype = [1,0,1]
e, f, v = dp.eval(coord, cell, atype)

The training data used is for Crystalline Silicon (432 atoms, 603 timesteps), provided online from the Deepmodeling repository.

The error I am receiving is the following:

Input to reshape is a tensor with 160 values, but the requested shape requires a multiple of 480

Could you help me with this issue?

Thank you very much!

**Comment by wanghan-iapcm** [comment]
It seems that you have only one atom type in your model. Please change `atype = [1,0,1]` to `atype = [0,0,0]`.

**Reply by npiroozan** [reply]
Thank you very much.  My apologies for not recognizing that myself.  I greatly appreciate your help.  I just have one final query.

Out of curiosity, is it possible to go beyond 3 sets of input coordinates?  What else would have to change to allow for say 10 sets of (x,y,z) coordinates as input to the inference model?

Thank you very much!

**Reply by njzjz** [reply]
Yes.

The shape of coordinates: (n_frames, n_atoms, 3)
The shape of cells: (n_frames, 3, 3) or None

**Reply by npiroozan** [reply]
Understood, so nparray().reshape(#cells,#atoms,3)

Thank you very much!

**Reply by npiroozan** [reply]
Please forgive me for one last question regarding the last point.  The following is the input script for x,y,z coordinates regarding 1 frame and 7 atoms.  The box size is 10x10x10.  All coordinates are within the box, nothing beyond the edges.


from deepmd.infer import DeepPot
import numpy as np
dp = DeepPot('graph.pb')
coord = np.array([[0,0,0], [0,2.71535,2.71535], [2.71535,0,2.71535], [2.71535,2.71535,0], [1.35767,1.35767,1.35767], [1.35767,4.07302,4.07302], [4.07302,1.35767,4.07302]]).reshape([1, 7, 3])

cell = np.diag(10 * np.ones(3)).reshape([1, 3, 3])
atype = [0,0,0]
e, f, v = dp.eval(coord, cell, atype)


The python script runs fine but there is an error within deepmd.infer:

_cannot reshape array of size 21 into shape (9)_


Apologies again, but what mistake am I making here?

Thank you very much.

**Reply by njzjz** [reply]
> atype = [0,0,0]

Only 3 atomic types are given but you have 7 atoms.
